ogc/edr/edr_provider.py

```
@@ -0,0 +1,745 @@
+
+    @staticmethod
+    def get_datetimes(layers: List[pogc.Layer]) -> List[str]:
+        """The list of available times for the provided layers.
+
+        Parameters
+        ----------
+        layers : List[pogc.Layer]
+            The list of layers to determine datetimes for.
+
+        Returns
+        -------
+        List[str]
+            Available time strings for the provider layers.
+        """
+
+        available_times = set()
+        for layer in layers:
+            if hasattr(layer, "valid_times") and layer.valid_times is not tl.Undefined and len(layer.valid_times) > 0:
+                available_times.update(layer.valid_times)
+
+        return list(available_times)
+
+    @staticmethod
+    def interpret_crs(crs: str | None) -> str:
+        """Interpret the CRS id string into a valid PyProj CRS format.
+
+        If None provided or the CRS is unknown, return the default.
+
+        Parameters
+        ----------
+        crs : str
+            The input CRS id string which needs to be validated/converted.
+
+        Returns
+        -------
+        str
+            Pyproj CRS string.
+        """
+        default_crs = "urn:ogc:def:crs:OGC:1.3:CRS84"  # Pyproj acceptable format
+        if crs is None or crs.lower() == "crs:84" or crs.lower() not in settings.EDR_CRS.keys():
+            return default_crs
+
+        return crs
+
+    @staticmethod
+    def crs_converter(x: Any, y: Any, crs: str) -> Tuple[Any, Any]:
+        """Convert the X, Y data to Longitude, Latitude data with the provided crs.
+
+        Parameters
+        ----------
+        x : Any
+            X data in any form.
+        y: Any
+            Y data in any form.
+        crs : str
+            The input CRS id string to apply to convert the X,Y data.
+
+        Returns
+        -------
+        Tuple[Any, Any]
+            The X,Y as Longitude/Latitude data.
+        """
+        if crs.lower() == "epsg:4326":
+            return (y, x)
+
+        return (x, y)
```

Is this a case where you can get away with raising an exception for an unsupported CRS?  A silent failure could be hard to debug

---

I will change the interpret_crs function to raise exceptions on unsupported types. This will align with EDR standard too.